src/include/autoinstall/ask.rb

```diff
@@ -261,35 +269,47 @@ def askDialog
               end
             end
           end
-          if frametitle != ""
-            if frameBuffer == nil
-              frameBufferVBox = VBox(dlg)
-            else
-              if frametitle == frameBufferTitle
-                frameBufferVBox = Builtins.add(frameBufferVBox, dlg)
-              else
-                dialog_term = Builtins.add(dialog_term, frameBuffer)
+          #
+          # At this time, widgets are created.
+          #
+
+          #
+          # Add the widget to the correct frame (using the frametitle).
+          #
+          if frametitle != "" # some frametitle is set
+            if frameBuffer == nil # and no frameBuffer exists
+              frameBufferVBox = VBox(dlg) # create a new frameBuffer
+            else # a frameBuffer exists
+              if frametitle == frameBufferTitle # with same title
+                frameBufferVBox = Builtins.add(frameBufferVBox, dlg) # add current to frameBuffer
+              else # with different title
+                dialog_term = Builtins.add(dialog_term, frameBuffer) # add frameBuffer to dialog
                 dialog_term = Builtins.add(dialog_term, VSpacing(1))
-                frameBufferVBox = VBox(dlg)
+                frameBufferVBox = VBox(dlg) # populate the frameBuffer with the new frame
               end
             end
-            frameBuffer = Frame(frametitle, frameBufferVBox)
+            frameBuffer = Frame(frametitle, frameBufferVBox) # set frameBuffer values for the next iteration
             frameBufferTitle = frametitle
-          else
-            if frameBuffer != nil
-              dialog_term = Builtins.add(dialog_term, frameBuffer)
+          else # frametitle is in blank
+            if frameBuffer != nil # a previous frameBuffer exists
```

Comment:
Yes, sure. Anyway, I only added comments to that part (I didn't want to change a single line of code except for adding the new feature).